copy-dlls-to-unity.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+set -x
+
+cp ./JsonFx/JsonFx.Json.UnitTests/bin/Release/net461/JsonFx.Json.dll ./JsonFx-Unity/Assets/Tests/Editor/JsonFx.Json.dll
```

Comment:
If you're going use the current directory, when you really to be in the same directory as the script, there's a magic line you can use to achieve this:

```
cd "$(dirname "$0")"
```
to break it down:
$0 is a special args variable that returns the full name of the invoking process. This is the script name. If it was invoked from the current directory, it will be ./copy-dlls-to-unity.sh or workspace/jsonfx-for-unity3d/copy-dlls-to-unity.sh

putting it in "" quotes will prevent any whitespace in its name from being interpretted as separate arguments

`dirname "$0"` dirname is a bit like Path.GetDirectoryName, it will return "." for "./copy-dlls-to-unity.sh" (or "copy-dlls-to-unity.sh") as input or "../random/workspace/jsonfx-for-unity3d/" for "../random/workspace/jsonfx-for-unity3d/copy-dlls-to-unity.sh" as input.

the `$()` (a more modern syntax of the ``` `` ```) means evaluate the script inside and return the output from the standard output as arguments wherever it is.

Then we put it in double quotes "" again to preserve whitespace.

sorry maybe this is TMI 🙈

Reply:
no not TMI! Thank you! You can smell my novice scripting a mile away :D